From d8eb0f5704f34cb7d411cd275d32c63ead297b8d Mon Sep 17 00:00:00 2001
From: Sumit Bose <sbose@redhat.com>
Date: Wed, 24 Aug 2016 15:37:41 +0200
Subject: [PATCH 01/23] Remove upper-case only check when looking for the
 NetBIOS name

It is a convention to use only upper-case letters for NetBIOS names but
it is not enforced on the AD-side. With the new option to specify a
random NetBIOS name it is possible to create host entries in AD with
lower-case letters in the name. To properly determine the name from the
keytab the upper-case check should be dropped,dc=
---
 library/adenroll.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/library/adenroll.c b/library/adenroll.c
index a15e4be..d1020e9 100644
--- a/library/adenroll.c
+++ b/library/adenroll.c
@@ -1309,7 +1309,7 @@ load_keytab_entry (krb5_context k5,
 	if (!enroll->host_fqdn_explicit && !enroll->computer_name_explicit) {
 
 		/* Automatically use the netbios name */
-		if (!enroll->computer_name && len > 1 && _adcli_str_is_up (name) &&
+		if (!enroll->computer_name && len > 1 &&
 		    _adcli_str_has_suffix (name, "$") && !strchr (name, '/')) {
 			enroll->computer_name = name;
 			name[len - 1] = '\0';
-- 
2.14.4

